index.js

```diff
@@ -206,30 +208,40 @@ module.exports = {
         }
 
         const diff = new PNG({ width: baseImg.width, height: baseImg.height });
-        const errorPixelCount = pixelmatch(baseImg.data, tmpImg.data, diff.data, baseImg.width, baseImg.height, {
-          threshold: options.imageMatchThreshold,
-          includeAA: options.includeAA
-        });
-
-        if (errorPixelCount <= options.imageMatchAllowedFailures) {
-          return resolve();
-        }
-
-        let diffPath = path.join(options.imageDiffDirectory, fileName);
-
-        await fs.outputFile(diffPath, PNG.sync.write(diff));
-
-        Promise.all([
-          _this._tryUploadToImgur(imgPath),
-          _this._tryUploadToImgur(diffPath)
-        ]).then(([urlTmp, urlDiff]) => {
+        try {
+          const errorPixelCount = pixelmatch(baseImg.data, tmpImg.data, diff.data, baseImg.width, baseImg.height, {
+              threshold: options.imageMatchThreshold,
+              includeAA: options.includeAA
+            });
+
+            if (errorPixelCount <= options.imageMatchAllowedFailures) {
+              return resolve();
+            }
+
+            let diffPath = path.join(options.imageDiffDirectory, fileName);
+
+            await fs.outputFile(diffPath, PNG.sync.write(diff));
+
+            Promise.all([
+              _this._tryUploadToImgur(imgPath),
+              _this._tryUploadToImgur(diffPath)
+            ]).then(([urlTmp, urlDiff]) => {
+              reject({
+                errorPixelCount,
+                allowedErrorPixelCount: options.imageMatchAllowedFailures,
+                diffPath: urlDiff || diffPath,
+                tmpPath: urlTmp || imgPath
+              });
+            }).catch(reject);
+        } catch (e) {
+          if(e.message !== IMAGE_SIZE_ERROR) console.error(e)
           reject({
-            errorPixelCount,
+            errorPixelCount: Math.abs((baseImg.data.length - tmpImg.data.length) / 4),
```

Comment:
Is this correct? 🤔 Can you explain this?

Reply:
ah I see, this is virtual here basically.. Can you rewrite this to make this clearer to something like this:

```js
} catch (e) {
  if(e.message === IMAGE_SIZE_ERROR) {
    return reject(...);
  }
  reject(e); 
}
```

Currently this would swallow other errors, I think?

Reply:
image.data.length is the number of pixels x 4 (for RGBA channels); within the version of Pixelmatch this is using, it verifies that the image data is exactly width x height x 4.
Taking the absolute value will return a positive count of differing pixels, regardless of which image is smaller.